tgui/packages/tgui/interfaces/HairPicker.tsx

```diff
@@ -0,0 +1,226 @@
+import { hexToHsva, HsvaColor, hsvaToHex } from 'common/color';
+import { BooleanLike } from 'common/react';
+import { createRef, useState } from 'react';
+
+import { useBackend } from '../backend';
+import {
+  Box,
+  Button,
+  ColorBox,
+  DmIcon,
+  Dropdown,
+  Modal,
+  Section,
+  Stack,
+  Tooltip,
+} from '../components';
+import { Window } from '../layouts';
+import { ColorSelector } from './ColorPickerModal';
+
+type HairPickerData = {
+  hair_icon: string;
+  hair_styles: { name: string; icon: string }[];
+  hair_style: string;
+  hair_color: string;
+
+  facial_hair_icon: string;
+  facial_hair_styles: { name: string; icon: string }[];
+  facial_hair_style: string;
+  facial_hair_color: string;
+
+  gradient_available: BooleanLike;
+  gradient_styles: string[];
+  gradient_style: string;
+  gradient_color: string;
+};
+
+export const HairPicker = () => {
+  const { act, data } = useBackend<HairPickerData>();
+
+  const {
+    hair_icon,
+    hair_style,
+    hair_styles,
+    hair_color,
+    facial_hair_icon,
+    facial_hair_style,
+    facial_hair_styles,
+    facial_hair_color,
+    gradient_available,
+    gradient_style,
+    gradient_styles,
+    gradient_color,
+  } = data;
+
+  const [colorPicker, setColorPicker] = useState<
+    'hair' | 'facial_hair' | 'gradient' | false
+  >(false);
+
+  let height = 340;
+  if (facial_hair_styles.length > 1) {
+    height = height + 300;
+  }
+
+  if (gradient_available) {
+    height = height + 85;
+  }
+
+  let defaultColor = '#000000';
+  switch (colorPicker) {
+    case 'hair':
+      defaultColor = hair_color;
+      break;
+    case 'facial_hair':
+      defaultColor = facial_hair_color;
+      break;
+    case 'gradient':
+      defaultColor = gradient_color;
+      break;
+
+    default:
+      break;
+  }
+
+  return (
+    <Window width={400} height={height} theme={'crtblue'}>
+      <Window.Content className="HairPicker">
+        <PickerElement
+          name="Hair"
+          icon={hair_icon}
+          hair={hair_styles}
+          active={hair_style}
+          color={hair_color}
+          setColor={setColorPicker}
+          action="hair"
+        />
+        {!!(facial_hair_styles.length > 1) && (
+          <PickerElement
+            name="Facial Hair"
+            icon={facial_hair_icon}
+            hair={facial_hair_styles}
+            active={facial_hair_style}
+            color={facial_hair_color}
+            setColor={setColorPicker}
+            action="facial_hair"
+          />
+        )}
+        {gradient_available && (
```

Comment:
```suggestion
        {!!gradient_available && (
```
This also was making 0s